io_gripper_controller/src/io_gripper_controller.cpp

```diff
@@ -0,0 +1,1376 @@
+      if (config_index_ == configurations_list_.end())
+      {
+        RCLCPP_ERROR(get_node()->get_logger(), "Configuration not found");
+        reconfigure_state_buffer_.writeFromNonRT(reconfigure_state_type::IDLE);
+        break;
+      }
+      else
+      {
+        conf_it_ = config_map_[std::distance(configurations_list_.begin(), config_index_)];
+        reconfigure_state_buffer_.writeFromNonRT(reconfigure_state_type::SET_COMMAND);
+      }
+      setResult = false;
+      for (const auto & io : conf_it_.command_high)
+      {
+        setResult = find_and_set_command(io, 1.0);
+        if (!setResult)
+        {
+          RCLCPP_ERROR(
+            get_node()->get_logger(), "Failed to set the command state for %s", io.c_str());
+        }
+      }
+      for (const auto & io : conf_it_.command_low)
+      {
+        setResult = find_and_set_command(io, 0.0);
+        if (!setResult)
+        {
+          RCLCPP_ERROR(
+            get_node()->get_logger(), "Failed to set the command state for %s", io.c_str());
+        }
+      }
+      reconfigure_state_buffer_.writeFromNonRT(reconfigure_state_type::CHECK_STATE);
+      break;
+
+    case reconfigure_state_type::CHECK_STATE:
+      check_state_ios_ = false;
+      for (const auto & io : conf_it_.state_high)
+      {
+        setResult = find_and_get_state(io, state_value_);
+        if (!setResult)
+        {
+          check_state_ios_ = false;
+          RCLCPP_ERROR(get_node()->get_logger(), "Failed to get the state for %s", io.c_str());
+        }
+        else
+        {
+          if (!(std::abs(state_value_ - 1.0) < std::numeric_limits<double>::epsilon()))
+          {
+            check_state_ios_ = false;
+            break;
+          }
+          else
+          {
+            check_state_ios_ = true;
+          }
+        }
+      }
+
+      for (const auto & io : conf_it_.state_low)
+      {
+        setResult = find_and_get_state(io, state_value_);
+        if (!setResult)
+        {
+          RCLCPP_ERROR(get_node()->get_logger(), "Failed to get the state for %s", io.c_str());
+        }
+        else
+        {
+          if (!(std::abs(state_value_ - 0.0) < std::numeric_limits<double>::epsilon()))
+          {
+            check_state_ios_ = false;
+            break;
+          }
+          else
+          {
+            check_state_ios_ = true;
+          }
+        }
+      }
+
+      if (check_state_ios_)
+      {
+        for (size_t i = 0; i < conf_it_.joint_states.size(); ++i)
+        {
+          joint_state_values_[i + params_.open_close_joints.size()] = conf_it_.joint_states[i];
+        }
+        reconfigure_state_buffer_.writeFromNonRT(reconfigure_state_type::IDLE);
+        configure_gripper_buffer_.writeFromNonRT("");
+        reconfigureFlag_.store(false);
+      }
+      break;
+    default:
+      break;
+  }
+}
+
+controller_interface::CallbackReturn IOGripperController::check_parameters()
```

Comment:
Please update the tests according to these checks of the parameters I have changed below. We should be able to work with parameter size 0.